### PR TITLE
[NTUSER][USER32] Set WM_CONTEXTMENU's wParam to the child window's handle

### PR DESCRIPTION
**Pull request by roytam1**
## Purpose

Set WM_CONTEXTMENU's wParam to the child window's handle

JIRA issue: [CORE-18801](https://jira.reactos.org/browse/CORE-18801)

## Proposed changes

Cherry-picked from https://github.com/wine-mirror/wine/commit/3af8415ca9dc50e6c394c1001aad97db5f514069

### CODE REVIEW

**Comment by roytam1** [comment]
please hold on. just tested with artifacts but I can't get desired result:

````
trace:(/win32ss/user/user32/windows/spy.c:2636)     (00030138)  DefWindowProc:[007b] WM_CONTEXTMENU  wp=00030138 lp=00f302b3
trace:(/win32ss/user/user32/windows/spy.c:2636)         (00030136)  DefWindowProc:[007b] WM_CONTEXTMENU  wp=00030138 lp=00f302b3
trace:(/win32ss/user/user32/windows/spy.c:2636)             (0003013C)  DefWindowProc:[007b] WM_CONTEXTMENU  wp=00030138 lp=00f302b3
trace:(/win32ss/user/user32/windows/spy.c:2675)              (0003013C)  DefWindowProc: [007b] WM_CONTEXTMENU returned 00000000
trace:(/win32ss/user/user32/windows/spy.c:2675)          (00030136)  DefWindowProc: [007b] WM_CONTEXTMENU returned 00000000
trace:(/win32ss/user/user32/windows/spy.c:2675)      (00030138)  DefWindowProc: [007b] WM_CONTEXTMENU returned 00000000
````
image

**Comment by roytam1** [comment]
OK it is working as desired now. Please review!

File: modules/rostests/winetests/user32/msg.c [hunk 1]
```diff
@@ -15569,6 +15577,23 @@ static void test_defwinproc(void)
...
+    flush_sequence();
+    test_context_menu = TRUE;
+    DefWindowProcA(child[2], WM_CONTEXTMENU, 0xcafe, 0);
+    test_context_menu = FALSE;
+    WmContextMenuSeq[0].wParam = (WPARAM)child[2];
+    WmContextMenuSeq[1].wParam = (WPARAM)child[1];
+    WmContextMenuSeq[2].wParam = (WPARAM)child[0];
+    ok_sequence(WmContextMenuSeq, "DefWindowProcA(WM_CONTEXTMENU)", FALSE);
+    DestroyWindow(child[0]);
+
```
Comment (binarymaster) [inline modules/rostests/winetests/user32/msg.c:15596]: It would be better to have separate commit for API test changes.

Comment (roytam1) [reply modules/rostests/winetests/user32/msg.c:15596]: > separate commit for API test changes

sure!

**Review by katahiromz — APPROVED** [review]

File: win32ss/user/user32/windows/defwnd.c [hunk 2]
```diff
@@ -390,11 +390,11 @@ User32DefWindowProc(HWND hWnd,
             {
                 if (bUnicode)
                 {
-                    SendMessageW(GetParent(hWnd), Msg, wParam, lParam);
+                    SendMessageW(GetParent(hWnd), Msg, (WPARAM)hWnd, lParam);
                 }
                 else
                 {
-                    SendMessageA(GetParent(hWnd), WM_CONTEXTMENU, wParam, lParam);
+                    SendMessageA(GetParent(hWnd), WM_CONTEXTMENU, (WPARAM)hWnd, lParam);
```
Comment (HBelusca) [inline win32ss/user/user32/windows/defwnd.c:397]: Here, the second parameter could be "`Msg`" as in the unicode case.

Comment (roytam1) [reply win32ss/user/user32/windows/defwnd.c]: sure

**Review by HBelusca — APPROVED** [review]

**Review by binarymaster — APPROVED** [review]
